drivers/include/sps30.h

```diff
@@ -0,0 +1,287 @@
+ *
+ * @param[in]   dev               Pointer to an SPS30 device handle
+ *
+ * @return      #SPS30_OK on success, negative #sps30_error_code_t on error
+ */
+int sps30_stop_measurement(const sps30_t *dev);
+
+/**
+ * @brief       Ask the device if a measurement is ready for reading.
+ *
+ * @param[in]   dev        Pointer to an SPS30 device handle
+ * @param[out]  error      Pre-allocated memory to return #sps30_error_code_t
+ *                         or NULL if not interested
+ *
+ * @return      true       if a new measurement is available
+ * @return      false      if no new measurement is available
+ */
+bool sps30_data_ready(const sps30_t *dev, int *error);
+
+/**
+ * @brief       Read a set of particulate matter measurements.
+ *
+ * @param[in]   dev        Pointer to an SPS30 device handle
+ * @param[out]  data       Pre-allocated memory to return measurements
+ *
+ * @return      #SPS30_OK on success, negative #sps30_error_code_t on error
+ */
+int sps30_read_measurement(const sps30_t *dev, sps30_data_t *data);
+
+/**
+ * @brief       Read the fan auto-clean interval.
+ *
+ * @details     The default value is 604800 seconds (1 week).
+ *              See also @ref sps30_start_fan_clean.
+ *
+ * @param[in]   dev        Pointer to an SPS30 device handle
+ * @param[out]  seconds    Pre-allocated memory for returning the interval,
+ *                         0 stands for disabled auto-clean
+ *
+ * @return      #SPS30_OK on success, negative #sps30_error_code_t on error
+ */
+int sps30_read_ac_interval(const sps30_t *dev, uint32_t *seconds);
+
+/**
+ * @brief       Write the fan auto-clean interval.
+ *
+ * @details     The new value will be effective immediately after writing but
+ *              reading the updated value is only possible after resetting the
+ *              sensor.
+ *              This setting is persistent across resets and powerdowns. But if
+ *              the sensor is powered off, the active time counter starts from
+ *              zero again. If this is expected to happen, a fan cleaning cycle
+ *              should be triggered manually at least once a week.
+ *              See also @ref sps30_start_fan_clean.
+ *
+ * @param[in]   dev        Pointer to an SPS30 device handle
+ * @param[out]  seconds    The new interval in seconds, 0 for disable
+ *
+ * @return      #SPS30_OK on success, negative #sps30_error_code_t on error
+ */
+int sps30_write_ac_interval(const sps30_t *dev, uint32_t seconds);
+
+/**
+ * @brief       Run a fan cleaning cycle manually.
+ *
+ * @details     This will spin up the fan to maximum speed to blow out dust for
+ *              for 10 seconds. No new measurement values are available until
+ *              the cleaning process is finished.
+ *
+ * @param[in]   dev        Pointer to an SPS30 device handle
+ *
+ * @return      #SPS30_OK on success, negative #sps30_error_code_t on error
+ */
+int sps30_start_fan_clean(const sps30_t *dev);
```

Comment:
nit picky: there is no `sps30_stop_fan_clean` 😮

Reply:
Yep because that is not a feature documented by the manufacturer. It could be possible to implement that via the reset command (not tested) but then this is a *very* unlikely thing to be needed. The clean will be done after a fixed time of 10 seconds anyway. And what kind of firmware would first trigger the clean and then decide it is better to stop it now?^^

Reply:
yeah sure, was just bike-shedding 😄 

If any I would change this to: `sps30_clean_fan` or even a simple `sps30_clean`.

Reply:
ahh I leave it as is. I actually chose the name like that because directly maps to how the command is named in the datasheet.